load_game.py

```diff
@@ -20,8 +23,97 @@
 FFXC = xbox.controller_handle()
 game_vars = vars.vars_handle()
 
-
-def load_into_game(gamestate: str, step_counter: str):
+def move_after_load(spec_move:str):
+    if spec_move == "miihen_laugh":
+        load_miihen_start_laugh()
+    elif spec_move == "miihen_no_laugh":
+        load_miihen_start()
+    elif spec_move == "MRR":
+        load_mrr()
+    
+
+def load_into_game(gamestate:str, step_counter:str):
+    logger.debug(f"Loading game state {gamestate} | {step_counter}")
+    # If wrong maps are loaded in, try to reset.
+    if memory.main.get_map() not in [23, 348, 349]:
+        reset.reset_to_main_menu()
+    area.dream_zan.new_game(gamestate=gamestate)
+    
+    # Now to get details for the load/save files
+    filepath = "json_ai_files\\save_load_details.json"
+    with open(filepath, "r") as fp:
+        results = json.load(fp)
+    
+    # Try to use new method, otherwise try old method.
+    try:
+        step_counter = str(step_counter)
+        num_results = len(results[gamestate][step_counter])
+        if num_results == 0:
+            logger.debug("Failure 1")
+            load_into_game_old(gamestate=gamestate, step_counter=step_counter)
+            return
+        
+        # Init variables so we don't crash later
+        save_num_conf = 0
+        nemesis_conf = "none"
+        blitz_win = "none"
+        end_ver = "none"
+        nea_zone = "none"
+        nem_ap = "none"
+        spec_move = "none"
+        
+        print(results[gamestate][step_counter].keys())
+        for key in results[gamestate][step_counter]:
+            save_num = int(results[gamestate][step_counter][key]["save_num"])
+            
+            nemesis = key
+            if save_num > 200:
+                pass
+            elif nemesis == str(game_vars.nemesis()):
+                logger.debug(f"Found save {save_num}")
+                save_num_conf = save_num
+                nemesis_conf = nemesis
+                blitz_win = results[gamestate][step_counter][key]["blitz_win_value"]
+                end_ver = results[gamestate][step_counter][key]["end_game_version_val"]
+                nea_zone = results[gamestate][step_counter][key]["nea_zone"]
+                nem_ap = results[gamestate][step_counter][key]["nem_ap_val"]
+                spec_move = results[gamestate][step_counter][key]["special_movement"]
+                logger.debug(f"Blitz Win {blitz_win}")
+                logger.debug(f"End game version {end_ver}")
+                logger.debug(f"NEA zone {nea_zone}")
+                logger.debug(f"Nemesis checkpoint {nem_ap}")
+        
+        if save_num_conf == 0:
+            logger.debug("Failure 2")
+            load_into_game_old(gamestate=gamestate, step_counter=step_counter)
+            return
+        else:
+            # Perform the load
+            load_save_num(int(save_num_conf))
+            game_vars.set_blitz_win(value=(blitz_win=="True"))
```

Comment:
Is there a reason we converted everything to string in JSON dump?

Reply:
Eh, retains the capital letter. In my mind it was easier to do quick string compares vs boolean comparisons across profiles.

Reply:
Assuming we are `json.load`ing the JSON file and these were not all strings, this line could just be `game_vars.set_blitz_win(value=blitz_win)` Which I feel like is cleaner.